src/entity/Unit.ts

```diff
@@ -763,7 +763,9 @@ export function die(unit: IUnit, underworld: Underworld, prediction: boolean) {
   // Note: This must come AFTER onDeathEvents or else it will remove the modifier
   // that added the onDeathEvent and the onDeathEvent won't trigger
   for (let [modifier, _modifierProperties] of Object.entries(unit.modifiers)) {
```

Comment:
A leading "_" denotes that the variable is unused.  What's happening here is called "Destructuring" and it works with both objects and arrays:
```
let a = [0,1,2,3];
let [zero, one] = a;
```
```
let a = {health:10,mana:2};
let {health, mana} = a;
```
In this example, I was just extracting the modifier key and the properties were unused.   Since the properties are now used, it is conventional to remove the leading "_"